Apps/Examples/Examples/All Examples/DebugMapExample.swift

```diff
@@ -0,0 +1,179 @@
+        MapDebugOptionSetting(debugOption: .stencilClip, displayTitle: "Show stencil buffer"),
+        MapDebugOptionSetting(debugOption: .tileBorders, displayTitle: "Debug tile clipping"),
+        MapDebugOptionSetting(debugOption: .timestamps, displayTitle: "Show tile loaded time"),
+    ]
+
+    init(debugOptions: [MapDebugOptions]) {
+        enabledDebugOptions = debugOptions
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        listView = UITableView()
+        listView.dataSource = self
+
+        view.addSubview(listView)
+
+        listView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            listView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            listView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+            listView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            listView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
+        ])
+
+        navigationItem.largeTitleDisplayMode = .never
+        navigationItem.rightBarButtonItem = UIBarButtonItem(
+            barButtonSystemItem: .save,
+            target: self,
+            action: #selector(saveSettings(_:)))
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        preferredContentSize = listView.contentSize
+    }
+
+    @objc private func saveSettings(_ sender: UIBarButtonItem) {
+        delegate?.debugOptionSettingsDidChange(self)
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        allSettings.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellID = String(describing: DebugOptionCell.self)
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellID) as? DebugOptionCell ?? DebugOptionCell()
+
+        let setting = allSettings[indexPath.row]
+        cell.configure(with: setting, isOptionEnabled: enabledDebugOptions.contains(setting.debugOption))
+        cell.onToggled { [unowned self] isEnabled in
+            if !isEnabled {
+                self.enabledDebugOptions.removeAll(where: { $0 == setting.debugOption })
+            } else if !self.enabledDebugOptions.contains(setting.debugOption) {
+                self.enabledDebugOptions.append(setting.debugOption)
+            }
+        }
+
+        return cell
+    }
+}
+
+// MARK: Cell
+
+private class DebugOptionCell: UITableViewCell {
+
+    private let titleLabel = UILabel()
+    private let toggle = UISwitch()
+
+    init() {
+        super.init(style: .default, reuseIdentifier: String(describing: Self.self))
```

Comment:
To do https://github.com/mapbox/mapbox-maps-ios/pull/1225/files#r838898209, you'll probably also need to override `init(style:reuseIdentifier:)` instead since it's the class's designated initializer and will likely be called by UITableView when it needs a new instance.